pkg/controller/installation/core_controller.go

```diff
@@ -945,73 +944,26 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 		}
 	}
 
-	var managerInternalTLSSecret *corev1.Secret
-	managerInternalTLSSecret, err = utils.ValidateCertPair(r.client,
-		common.CalicoNamespace,
-		render.ManagerInternalTLSSecretName,
-		render.ManagerInternalSecretKeyName,
-		render.ManagerInternalSecretCertName,
-	)
-
-	// Ensure that CA and TLS certificate for tigera-manager for internal
-	// traffic within the K8s cluster exists and has valid FQDN manager service
-	// names and localhost.
+	certificateManager, err := certificatemanagement.CreateCertificateManager(r.client, &instance.Spec, r.clusterDomain)
+	if err != nil {
+		log.Error(err, "unable to create the Tigera CA")
+		r.status.SetDegraded("unable to create the Tigera CA", err.Error())
+		return reconcile.Result{}, err
+	}
+	var managerInternalTLSSecret certificatemanagement.KeyPair
 	if instance.Spec.Variant == operator.TigeraSecureEnterprise && managementCluster != nil {
-		var err error
-		svcDNSNames := dns.GetServiceDNSNames(render.ManagerServiceName, render.ManagerNamespace, r.clusterDomain)
-		svcDNSNames = append(svcDNSNames, render.ManagerServiceIP)
-		certDur := 825 * 24 * time.Hour // 825days*24hours: Create cert with a max expiration that macOS 10.15 will accept
-
-		managerInternalTLSSecret, _, err = utils.EnsureCertificateSecret(
-			render.ManagerInternalTLSSecretName, managerInternalTLSSecret, render.ManagerInternalSecretKeyName, render.ManagerInternalSecretCertName, certDur, svcDNSNames...,
-		)
-
+		managerInternalTLSSecret, err = certificateManager.GetOrCreateKeyPair(r.client, render.ManagerInternalTLSSecretName, common.CalicoNamespace, append(dns.GetServiceDNSNames(render.ManagerServiceName, render.ManagerNamespace, r.clusterDomain), render.ManagerServiceIP))
```

Comment:
Could you keep the dns names as a variable, or make this two lines please. It's a very wide line which makes this more difficult to read (especially in a PR).